### PR TITLE
tests: verify container deletion while SN is down

### LINKED ISSUE #696: [NEW TEST]: Container deletion
## Context

We have solved some things in the node: https://github.com/nspcc-dev/neofs-node/issues/1663. The issue does not contain the full implementation, for more details, it would be better to see the discussions in the related PR.

### It was before

If a container is removed, its objects are not removed instantly. The policer tries to check if an object is stored well and finds that its container does not exist and, therefore, deletes the object. The policer works asyncly always. If you are lucky enough and the container is cached (so can't be fetched from the chain with 404 error) you can get that object (cache is usually 30s if enabled). But even if the cache is disabled or the container is missing, you probably get smth like ACL error with the missing container error but not the NeoFS error about missing/deleted objects.

### It is now

Containers are fetched in the initial stage, so "missing" objects are marked correspondingly on startup. ContainerDeleted event is being listened to in runtime so container deletion should be handled in runtime by the storage engine. The metabase has been expanded and it indexes stored objects that belong to the deleted container, it should be impossible to fetch such objects OR get a status error other than [object not found](https://github.com/nspcc-dev/neofs-api/blob/cae99c9c6328250666f9e6944166cd8a4cc54cf8/status/types.proto#L114).

## Test that could fix and check that behavior

1. Create a container
2. Put objects to it
3. Down some nodes that store the objects (but not all of them)
4. Delete the container
5. Up the nodes from the `3.`
6. Try to fetch the objects as fast as possible after the `4.` is done
7. Check the errors: only [not found](https://github.com/nspcc-dev/neofs-api/blob/cae99c9c6328250666f9e6944166cd8a4cc54cf8/status/types.proto#L114) are expected

If it is possible, 4-7 should be done asyncly or as fast as possible.

### PR DESCRIPTION
**Pull request by evgeniiz321**
closes #696

### CODE REVIEW

**Comment by evgeniiz321** [comment]
@carpawell take a look please

**Review by carpawell — COMMENTED** [review]
I did not expect to see `CONTAINER_NOT_FOUND` and mentioned it in the issue but now i have realized that ACL checks are done before any engine's checks so missing container errors could not be skipped. However, I see two alternatives: 1. enable a long morph (container) cache and try to get an object then (I expect ACL checks pass and engine's object not found error then (_however_, the cache can be cleared on container removal, do not remember it for sure)), 2. extend `neofs-cli control` command to be ready to check object statuses without any network rules, just simple engine's call.
2. can be tricky for the current test code (i am not sure), 1. will be not early than 0.43.0 release.

Currently, the test is valid but it does not check what i expected after the https://github.com/nspcc-dev/neofs-node/pull/2638 (it just cannot do it now).

**Comment by cthulhu-rider** [comment]
@carpawell i think cache tricks wouldn't be stable: cache invalidation on removal event is the most expected behavior, the node must deny removed container's requests asap. So, Control service call sounds quite generally useful

**Comment by carpawell** [comment]
> @carpawell i think cache tricks wouldn't be stable

Me too.

> So, Control service call sounds quite generally useful

Do not see another good solution here too: https://github.com/nspcc-dev/neofs-node/issues/2886.

File: pytest_tests/lib/helpers/grpc_responses.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
 NOT_SESSION_CONTAINER_OWNER = "session issuer differs with the container owner"
 TIMED_OUT = "timed out after \\d+ seconds"
 CONTAINER_DELETION_TIMED_OUT = "container deletion: await timeout expired"
+CONTAINER_NOT_FOUND = "code = 3072.*message = container not found"
```
[the remark this answers is not in the document]
Comment (evgeniiz321) [reply pytest_tests/lib/helpers/grpc_responses.py]: Fixed

File: pytest_tests/tests/container/test_container.py [hunk 2]
```diff
@@ -141,3 +176,46 @@ def test_container_creation_deletion_parallel(self, default_wallet):
...
+
+        with allure.step("Alive node should return GC MARKED for the created object from the deleted container"):
+            wait_until_gc_marked(self.neofs_env, alive_node_with_object, cid, oid)
+
+        with allure.step("Start storage node"):
+            node_to_stop.start(fresh=False)
+            wait_all_storage_nodes_returned(self.neofs_env)
+
+        with allure.step("Previously stopped node should return GC MARKED for the created object"):
+            wait_until_gc_marked(self.neofs_env, node_to_stop, cid, oid)
```
[the remark this answers is not in the document]
Comment (evgeniiz321) [reply pytest_tests/tests/container/test_container.py]: Fixed

**Review by carpawell — APPROVED** [review]

**Pull request merged**